### DISCUSSION TITLE
USWDS - Feature: Add a sticky variant for header navigation

### DISCUSSION THREAD

**Discussion by amyleadem** [uswds/uswds#6961] [Ideas]
## Summary
We should consider creating a variant for the header navigation component that is similar to the sticky header on [designsystem.digital.gov](https://designsystem.digital.gov/). 

## Resources
- Related to this [question in ZenDesk](https://gsa-solutionshelp.zendesk.com/agent/tickets/14331) (🔒 ).
- Related to this [Slack thread](https://gsa-tts.slack.com/archives/C3F14AHSQ/p1698932431822059).

**Comment by annepetersen** [comment]
The USWDS core team noted that we use a sticky header on our site and haven't run into usability issues with it in our testing, but we're curious how broad of a need this is or how much usage it might get. 

We'll plan to integrate this into our quarterly planning, and hopefully put it out as a discussion and solicit voting from the community at a time we can integrate the work for it.

In the meantime, we'll put it in our usability research backlog and plan to do an accessibility audit pass on it.

**Comment by ryparker** [comment]
I've moved this to Ideas to collect the examples requested in the earlier maintainer response. The original request and comments are preserved here; implementation isn't scheduled yet.

If your team needs sticky header navigation, please share the page or workflow, where the existing header falls short, and any findings from mobile, keyboard, zoom, or screen-reader testing. We'd still need the usability research and accessibility audit described above before supporting a new variant. [USAGov's mobile-menu discussion](https://github.com/uswds/uswds/discussions/6626) has related examples, though its broader menu changes are a separate topic. Those examples can help us define a focused implementation issue.
